app/models/graph/types/query.rb

```diff
@@ -46,6 +46,17 @@ module Types
         }
       end
 
+      field :viewer, Graph::Types::User, 'The currently authenticated user (if any)' do
+        resolve ->(_, _, ctx) {
+          case ctx[:user]
+          when ::User
```

Comment:
Do we need GuestUser if we just return nil in these cases ? I guess eventually you want to show certain fields under viewer even when youre a guest ?

I think i like the second approach better yep!